app/Livewire/TaStatsGeneral.php

```diff
@@ -0,0 +1,127 @@
+<?php
+
+namespace App\Livewire;
+
+use App\TrainingTicket;
+use Carbon\Carbon;
+use Livewire\Component;
+
+class TaStatsGeneral extends Component {
+    public $no_shows;
+    public $tickets_by_certs;
+    public $session_ids;
+    public $completion_ratios;
+    public \StdClass $lookback_monthly;
+    public \StdClass $lookback_annual;
+    private $cert_types = ['S1', 'S2', 'S3', 'C1'];
+    public $position_select;
+
+    public function render() {
+        $this->generate_stats();
+        return view('livewire.ta-stats-general');
+    }
+
+    private function generate_stats(): void {
+        $this->generate_no_shows();
+        $this->generate_tickets_by_certification();
+        $this->generate_completion_ratios();
+        $this->generate_lookback();
+    }
+
+    private function generate_no_shows(): void {
+        $by_controller = [];
+        $no_show_sessions = TrainingTicket::where('type', 10)->where('date', '<=', Carbon::now()->addDays(45)->toDateTimeString())->get();
+        foreach ($no_show_sessions as $no_show_session) {
+            $by_controller[] = $no_show_session->controller_id;
+        }
+        $more_than_one_no_show = array_count_values($by_controller);
+        if (count($more_than_one_no_show) == 0) {
+            $this->no_shows = [];
+            return;
+        }
+        if (max($more_than_one_no_show) < 2) {
+            $this->no_shows = [];
+            return;
+        }
+        foreach ($more_than_one_no_show as $controller => $number_no_shows) {
+            if ($number_no_shows < 2) {
+                continue;
+            }
+            $controller_no_shows = $no_show_sessions->where('controller_id', $controller)->get();
+            $controller_name = 'Unknown';
+            $controller_no_show_events = [];
+            foreach ($controller_no_shows as $no_show) {
+                $controller_no_show_events[] = $no_show->date;
+            }
+            $this->no_shows[] = $controller_name . ' no-showed sessions on the following dates: ' . implode(', ', $controller_no_show_events);
```

Comment:
Will this automatically append or does this overwrite the whole array?

Reply:
The intent of this code is to generate one string per controller which lists their recent no-shows. The code searches for no-shows that meet the criteria and then generate a string, adding the string to an array. `$this->no_shows[] = "bleh...";` should add each string to a new element of the array. The blade then uses a `@foreach` or the like to iterate through the array and display the individual strings.